src/stacks-hierarchy/Root_PurchaseOverviewScreen/Root_PurchaseOverviewScreen.tsx

```diff
@@ -34,6 +35,12 @@ export const Root_PurchaseOverviewScreen: React.FC<Props> = ({
 }) => {
   const styles = useStyles();
   const {t, language} = useTranslation();
+  const harborConnections = useHarbors(params.fromPlace?.id);
+
+  const toPlaceConnection = harborConnections.data.find(
+    (harbor) => harbor.id === params.toPlace?.id,
+  );
+  const isFree = !!toPlaceConnection?.isFree;
```

Comment:
The `isFree` flag isn't already on the `params.toPlace`? If not, and this is necessary, I think it would be appropriate to move it out into a `useIsFree` hook or something.

Reply:
It was moved to a useIsFree hook

Reply:
I don't fully understand why this part is necessary, since harbor connections returned from `useHarbors` should now already have the `isFree` flag 🤔 Could you take an extra look, and just contact if we should do a video session.

Reply:
Hi, let's have a talk ;)

Reply:
I made a refactor to the code after a talk in Slack ;)